packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -649,6 +649,22 @@ export class GarbageCollector implements IGarbageCollector {
          */
         this.initializeBaseStateP = new LazyPromise<void>(async () => {
             const currentReferenceTimestampMs = this.runtime.getCurrentReferenceTimestampMs();
+            /**
+             * If there is no current reference timestamp, skip initialization. We need the current timestamp to track
+             * how long objects have been unreferenced and if they can be deleted.
+             *
+             * Note that the only scenario where there is no reference timestamp is when no ops have ever been processed
+             * for this container and it is in read mode. In this scenario, there is no point in running GC anyway
+             * because references in the container do not change without any ops, i.e., there is nothing to collect.
+             */
+            if (currentReferenceTimestampMs === undefined) {
+                // Log an event so we can evaluate how often we run into this scenario.
+                this.mc.logger.sendErrorEvent({
+                    eventName: "GarbageCollectorInitializedWithoutTimestamp",
+                });
+                return;
```

Comment:
Maybe return `false` for these early returns?  Assert it's true in the `activeConnection` case...?  Could be over-aggressive, thinking out loud.  Also unsure about the conditions around the other early return(s)

Reply:
We can add these validations but the reasons I am not adding those are following:
- The most interesting case is when the summarizer client calls it. This always happens via the `submitSummary` funtion in container runtime. That function verifies that this is an active connection before it starts the summary / GC process.
- This keeps it somewhat in line with `summarize` method on container runtime. It does not have any validations for "write" mode. You can call it in absence of ops and it basically does nothing, i.e., everything is the same as before so it returns a bunch of ISummaryHandles. Similarly, you can call this without any ops and it would be a no-op. If you try to get the gc data (by calling summarize on garbage collector, you get nothing).

If we were to add validations, I would make `currentReferenceTimestampMs` a required parameter and let the creator / caller validate that it does not do so without any ops. However, I do not see a very good reason to add that complexity atm.

Reply:
I created an issue for adding assertions for write mode - https://dev.azure.com/fluidframework/internal/_workitems/edit/375